lib/oplog/tail.go

```diff
@@ -143,56 +172,137 @@ func (tailer *Tailer) tailOnce(out chan<- *redispub.Publication, stop <-chan boo
 
 		var rawData bson.Raw
 
-		for iter.Next(&rawData) {
-			ts, pubs := tailer.unmarshalEntry(rawData)
+		for {
+			gotResult, didTimeout, didLosePosition, err := readNextFromCursor(query)
 
-			if ts != nil {
-				lastTimestamp = *ts
-			}
+			if gotResult {
+				decodeErr := query.Decode(&rawData)
+				if decodeErr != nil {
+					log.Log.Errorw("Error decoding oplog entry", "error", decodeErr)
```

Comment:
do you have to return or otherwise handle the error here?

Reply:
I think logging the error and continuing is the best we can do here, especially if we get a single bad-apple unparseable oplog entry I wouldn't want it to bring down the service and block subsequent updates.

Reply:
I added a 'continue' just to be explicit